From 50d942a31fa2f89c4073a7813be83d7d1eafc8d2 Mon Sep 17 00:00:00 2001
From: Jack Yu <jackcwyu@google.com>
Date: Wed, 30 Jan 2019 15:35:01 +0800
Subject: [PATCH 3/5] Prevent OOB error in rw_i93_sm_read_ndef()

Bug: 122035770
Test: NFC tag reading
Change-Id: I63eccf903fedf048739c6b2bc663b3669b9ed69b
(cherry picked from commit a85ffb7b1adfd44202c9d958b3911727a59d5086)
---
 src/nfc/tags/rw_i93.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/nfc/tags/rw_i93.c b/src/nfc/tags/rw_i93.c
index e4a2aed..d883d96 100644
--- a/src/nfc/tags/rw_i93.c
+++ b/src/nfc/tags/rw_i93.c
@@ -1962,6 +1962,12 @@ void rw_i93_sm_read_ndef (BT_HDR *p_resp)
 
     RW_TRACE_DEBUG0 ("rw_i93_sm_read_ndef ()");
 
+    if (length == 0) {
+        android_errorWriteLog(0x534e4554, "122035770");
+        rw_i93_handle_error(NFC_STATUS_FAILED);
+        return;
+    }
+
     STREAM_TO_UINT8 (flags, p);
     length--;
 
-- 
2.7.4

